### PR TITLE
BUG: Empty lists shouldn't be counted as DateOffsets. See #13844

### LINKED ISSUE #13844: Empty dataset: typerror ufunc add cannot use operands with types dtype('<M8[ns]') and dtype('O')
#### Code Sample, a copy-pastable example if possible
```python
from pandas import DataFrame, to_datetime, to_timedelta


data = {'Item': [1], 'Date_Time': ['2/8/2015  6:00:30'], 'Relative_Time': 20}
dtf = DataFrame(data=data)

print(dtf)
dtf = dtf[dtf.Item == 2]  # if the filter returns empty dataset expected results will be that the same

dtf.Date_Time = to_datetime(dtf.Date_Time, infer_datetime_format=True)
dtf.Relative_Time = to_timedelta(dtf.Relative_Time, unit='s')
dtf.Date_Time = dtf.Date_Time + dtf.Relative_Time
print(dtf)
```
#### Current Output
```
          Date_Time  Item  Relative_Time
0  2/8/2015  6:00:30     1             20
/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/pandas/core/ops.py:477: PerformanceWarning: Adding/subtracting array of DateOffsets to Series not vectorized
  "Series not vectorized", PerformanceWarning)
Traceback (most recent call last):
  File "bug.py", line 12, in <module>
    dtf.Date_Time = dtf.Date_Time + dtf.Relative_Time
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/pandas/core/ops.py", line 641, in wrapper
    arr = na_op(lvalues, rvalues)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/pandas/core/ops.py", line 481, in <lambda>
    self.na_op = lambda x, y: getattr(x, self.name)(y)
TypeError: ufunc add cannot use operands with types dtype('<M8[ns]') and dtype('O')
```
#### Expected Output
```
          Date_Time  Item  Relative_Time
0  2/8/2015  6:00:30     1             20
Empty DataFrame
Columns: [Date_Time, Item, Relative_Time]
Index: []
```
#### output of ``pd.show_versions()``

```
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.1.final.0
python-bits: 64
OS: Darwin
OS-release: 15.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.18.1
nose: None
pip: 8.1.2
setuptools: 24.0.3
Cython: None
numpy: 1.11.1
scipy: 0.16.1
statsmodels: None
xarray: None
IPython: 4.0.1
sphinx: None
patsy: None
dateutil: 2.5.3
pytz: 2016.6
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.0
openpyxl: 2.3.5
xlrd: 1.0.0
xlwt: 1.0.0
xlsxwriter: None
lxml: None
bs4: 4.4.1
html5lib: None
httplib2: 0.9.2
apiclient: 1.5.0
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None
None
statsmodels: None
xarray: None
IPython: 4.0.1
sphinx: None
patsy: None
dateutil: 2.5.3
pytz: 2016.6
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.0
openpyxl: 2.3.5
xlrd: 1.0.0
xlwt: 1.0.0
xlsxwriter: None
lxml: None
bs4: 4.4.1
html5lib: None
httplib2: 0.9.2
apiclient: 1.5.0
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None
None
```

### PR DESCRIPTION
**Pull request by tom-bird**
- [x] closes #13844 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
tests!

**Comment by theultimatecrouton** [comment]
Ah, sorry. Could you point me in the direction of the relevant test file? The structure is different to numpy...

**Comment by jreback** [comment]
[contributing docs](http://pandas.pydata.org/pandas-docs/stable/contributing.html)

this is prob in ``tseries/tests_timeseries.py`` though maybe in ``tests/series/test_*`` (somewhere) as well. you will have to find something similar and put right after.

**Comment by theultimatecrouton** [comment]
Thanks. Right I've inserted a test in each of the files you mentioned. To be honest the logical distinction between the different classes in the test files wasn't that clear to me, so please say if they are in the wrong segment. 

First time contributing to pandas so just finding my feet!

File: pandas/core/ops.py [hunk 1]
```diff
@@ -598,7 +598,10 @@ def _is_offset(self, arr_or_obj):
         if isinstance(arr_or_obj, pd.DateOffset):
             return True
         elif is_list_like(arr_or_obj):
-            return all(isinstance(x, pd.DateOffset) for x in arr_or_obj)
+            if len(arr_or_obj):
+                return all(isinstance(x, pd.DateOffset) for x in arr_or_obj)
```
Comment (jreback) [inline pandas/core/ops.py]: you can just collapse this (meaning remove the ``else`` clauses and just ``return False`` at the end of the function

File: pandas/core/ops.py [hunk 2]
```diff
@@ -598,7 +598,10 @@ def _is_offset(self, arr_or_obj):
         if isinstance(arr_or_obj, pd.DateOffset):
             return True
         elif is_list_like(arr_or_obj):
-            return all(isinstance(x, pd.DateOffset) for x in arr_or_obj)
+            if len(arr_or_obj):
```
Comment (jreback) [inline pandas/core/ops.py]: maybe do this as an ``and``

File: pandas/tests/series/test_timeseries.py [hunk 3]
```diff
@@ -525,3 +525,8 @@ def test_timeseries_coercion(self):
             self.assertTrue(ser.is_time_series)
         self.assertTrue(ser.index.is_all_dates)
         self.assertIsInstance(ser.index, DatetimeIndex)
+
+    def test_empty_series_addition(self):
+        a = Series(dtype='M8[ns]')
```
Comment (jreback) [inline pandas/tests/series/test_timeseries.py]: add the issue number as a comment

File: pandas/tseries/tests/test_timeseries.py [hunk 4]
```diff
@@ -2243,6 +2243,11 @@ def test_timestamp_equality(self):
         result = s == s[1]
         assert_series_equal(result, Series([False, False]))
 
+    def test_empty_series_addition(self):
```
Comment (jreback) [inline pandas/tseries/tests/test_timeseries.py]: don't need to repeat this test here

File: pandas/tests/series/test_timeseries.py [hunk 5]
```diff
@@ -525,3 +525,8 @@ def test_timeseries_coercion(self):
...
+        b = Series(dtype='m8[ns]')
```
Comment (jreback) [inline pandas/tests/series/test_timeseries.py]: use ```tm.assert_series_equal`` ; construct the expected Series

File: pandas/tests/series/test_timeseries.py [hunk 6]
```diff
@@ -525,3 +525,8 @@ def test_timeseries_coercion(self):
...
+        assert_series_equal(a, a+b)
```
Comment (sinhrks) [inline pandas/tests/series/test_timeseries.py]: can u also add: ``b + a`` and ``a - b`` ?

File: pandas/tests/series/test_timeseries.py [hunk 7]
```diff
@@ -525,3 +525,11 @@ def test_timeseries_coercion(self):
...
         self.assertTrue(ser.index.is_all_dates)
         self.assertIsInstance(ser.index, DatetimeIndex)
+
+    def test_empty_series_addition(self):
+        # see issue #13844
+        a = Series(dtype='M8[ns]')
+        b = Series(dtype='m8[ns]')
+        assert_series_equal(a, a + b)
+        assert_series_equal(a, a - b)
+        assert_series_equal(a, b + a)
```
Comment (jreback) [inline pandas/tests/series/test_timeseries.py]: add the  ``b - a`` (which should raise)

File: pandas/tests/series/test_timeseries.py [hunk 8]
```diff
@@ -525,3 +525,11 @@ def test_timeseries_coercion(self):
             self.assertTrue(ser.is_time_series)
         self.assertTrue(ser.index.is_all_dates)
         self.assertIsInstance(ser.index, DatetimeIndex)
+
+    def test_empty_series_addition(self):
```
Comment (jreback) [inline pandas/tests/series/test_timeseries.py]: call this ``test_empty_series_ops``

**Comment by jreback** [comment]
minor points. ping on green.

**Comment by theultimatecrouton** [comment]
thanks, will amend

**Comment by jreback** [comment]
lgtm. pls add a whatsnew note in Bug Fixes.

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**